provider/oci/storage_volumes.go

```diff
@@ -534,6 +543,7 @@ func (v *volumeSource) DetachVolumes(ctx envcontext.ProviderCallContext, params
 		if !ok {
 			currentAttachments, err := v.volumeAttachments(param.InstanceId)
 			if err != nil {
```

Comment:
If there's a credential error here, there's no point continuing with the next loop below.
A call to MaybeHandleCredentialError will tell you if that's the case
(same with other loops in the storage area potentially, look at other providers for inspiration)

Reply:
I've added some logic to avoid unnecessary calls within that method, but haven't added anything similar to the rest of the storage methods yet.